tests/unit/test_charm.py

```diff
@@ -16,6 +16,30 @@
 from charm import NSSFOperatorCharm
 
 
+def get_tls_relation_with_certificates(certificate: str, csr: str):
+    """Returns a certificate Relation containing the given certificate and csr."""
+    tls_relation = Relation(
+        endpoint="certificates",
+        remote_app_name="tls-provider",
+        local_unit_data={
+            "certificate_signing_requests": json.dumps([{"certificate_signing_request": csr}])
+        },
+        remote_app_data={
+            "certificates": json.dumps(
+                [
+                    {
+                        "certificate": certificate,
+                        "certificate_signing_request": csr,
+                        "ca": "abc",
+                        "chain": ["abc", "def"],
+                    }
+                ]
+            )
+        },
+    )
+    return tls_relation
```

Comment:
We shouldn't have to manage relation data directly in unit tests as our code doesn't do this either, all it does is communicate with the `tls-certificates` lib. Here we should do the same and patch calls to this lib.